src/widget/wtracktableview.cpp

```diff
@@ -1775,6 +1785,33 @@ void WTrackTableView::doSortByColumn(int headerSection) {
     horizontalScrollBar()->setValue(savedHScrollBarPos);
 }
 
+void WTrackTableView::applySorting() {
```

Comment:
For the toggle feature, you need an extra slot for the "sort_column" CO.  
In case the coloumn changes sort_order should be reset to 0. If the sort column is unchanged, toggle sort_order.
The target is to have the same behavior like the header buttons .

Reply:
I added another control object `sort_column_toggle` that behaves like that. That made the implementation easier and allows more flexibility in case a mapping does not want the toggle behaviour.